### PR TITLE
Add support for Password Hardening

### PR DESCRIPTION
**Pull request by davidpil2002**
#### What I did
Add Password Hardening CLI
#### How I did it
created the CLI by using YANG model generator, the YANG model can be found in the password hardening HLD https://github.com/Azure/SONiC/blob/master/doc/passw_hardening/hld_password_hardening.md#TestPlan
and also in sonic-buildimage will be merged in the path: src/sonic-yang-models/yang-models/sonic-passwh.yang
#### How to verify it
Manually:
you can use configurations command like"config passw-hardening policies <policy> <value>"  or 
"show passw-hardening policies" (more examples in the HLD.)
Auto:
1.There are unitest of each policy including good & bad flow in this commit, that should pass.

2.There are tests in sonic-mgmt repo in the path: sonic-mgmt/tests/passw_hardening/
the test are end to end test and the are testing the config/show CLI commands as well.
#### Previous command output (if the output of a command-line utility has changed)
(new feature)
#### New command output (if the output of a command-line utility has changed)

```
root@r-panther-13:/home/admin# show passw-hardening policies
STATE    EXPIRATION    EXPIRATION WARNING    HISTORY    LEN MAX    LEN MIN    USERNAME PASSW MATCH    LOWER CLASS    UPPER CLASS    DIGITS CLASS    SPECIAL CLASS
-------  ------------  --------------------  ---------  ---------  ---------  ----------------------  -------------  -------------  --------------  ---------------
enabled      30           10                   4         100        30               false                 true            true            true          true

```

### CODE REVIEW

File: show/plugins/sonic-passwh_yang.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
+"""
+Auto-generated show CLI plugin.
+
+
+"""
+
+import click
+import tabulate
+import natsort
```
Comment (dgsudharsan) [inline show/plugins/sonic-passwh_yang.py:9]: Please remove natsort as it throws unused import.

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: tests/passw_hardening_test.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
```
Comment (dgsudharsan) [inline tests/passw_hardening_test.py:1]: Why do we need tests for auto-generated CLI? Is it because of the coverage?

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: tests/passw_hardening_test.py (hunk 2 above)
Comment (davidpil2002) [reply tests/passw_hardening_test.py:1]: yes

**Comment by liat-grozovik** [comment]
@liuh-80, @qiluo-msft kindly reminder to review and signoff

**Comment by qiluo-msft** [comment]
The latest coverage requirement (DIFF_COVER_CHECK_THRESHOLD) is 80%. Could you improve the coverage?

**Comment by zhangyanzhao** [comment]
Liat will check with Qi on the review.

File: config/plugins/sonic-passwh_yang.py [hunk 3]
```diff
@@ -0,0 +1,537 @@
+"""
+Autogenerated config CLI plugin.
```
[the remark this answers is not in the document]
Comment (davidpil2002) [reply config/plugins/sonic-passwh_yang.py]: Should be an HLD/DOC about how to use the tool, I removed the augenerated comment

**Comment by davidpil2002** [comment]
> The latest coverage requirement (DIFF_COVER_CHECK_THRESHOLD) is 80%. Could you improve the coverage?

I added more unitest and removed unused code.
the cov is now more than the threshold

**Comment by davidpil2002** [comment]
answered all the comment, the code is ready to be review and merge
@qiluo-msft 
@liat-grozovik

**Comment by qiluo-msft** [comment]
@liuh-80 Could you review this PR?

**Comment by davidpil2002** [comment]
@liat-grozovik pls can you help or derived me about this EasyCLA authorization issue?

**Comment by liat-grozovik** [comment]
@liuh-80 we need to close this feature review as it is for 202205. can you please help to approve the PR?

**Comment by liuh-80** [comment]
@liat-grozovik, please fix the easycla issue, then I can approve this PR.

**Review by liuh-80 — APPROVED** [review]
I will approve again after fix easycla issue.

**Comment by liuh-80** [comment]
> * ❌ The commit ([6b50c67](https://github.com/sonic-net/sonic-utilities/commit/6b50c6761154206d7c78af8bfb9dbde14985543f)). This user is missing the User's ID, preventing the EasyCLA check. [Consult GitHub Help](https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user) to resolve.For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

Please fix user name and emaill address issue on this commit:

❌ The commit (https://github.com/sonic-net/sonic-utilities/commit/6b50c6761154206d7c78af8bfb9dbde14985543f). This user is missing the User's ID, preventing the EasyCLA check. [Consult GitHub Help](https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user) to resolve.For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

**Comment by zhangyanzhao** [comment]
@davidpil2002 can you please sign the EasyCLA to unblock the merge process? Thanks. @liat-grozovik  for viz

**Comment by liuh-80** [comment]
@davidpil2002 , the CLA issue caused by this commit: https://github.com/sonic-net/sonic-utilities/commit/6b50c6761154206d7c78af8bfb9dbde14985543f

In this commit, your ID is 'davidpil'

In other commit which does not has the CLA issue, your ID is 'davidpil2002', so please check and fix your ID in that commit, then CLA will pass.

**Comment by davidpil2002** [comment]
> @davidpil2002 , the CLA issue caused by this commit: [6b50c67](https://github.com/sonic-net/sonic-utilities/commit/6b50c6761154206d7c78af8bfb9dbde14985543f)
> 
> In this commit, your ID is 'davidpil'
> 
> In other commit which does not has the CLA issue, your ID is 'davidpil2002', so please check and fix your ID in that commit, then CLA will pass.

I fixed this issue by creating a new PR with similar name:https://github.com/sonic-net/sonic-utilities/pull/2338

**Pull request closed without merging**